### PR TITLE
[AB#43320] check-finished-ingest-item message dropped

### PR DESCRIPTION
**Pull request by AxTrusov**
# Pull Request

## Prerequisites

- [x] The PR is targeting the right branch (`dev` for features and `master` for
      releases)
- [ ] potential **release notes** to the PR description added
- [x] potential **testing notes** to the PR description added
- [ ] appropriate labels for the PR applied

## Description

- `CheckFinishIngestItem` command dropped in favor of setting ingest item step state directly and making `CheckFinishIngestDocument` update ingest item states based on step states in a single update statement.
- `app_hidden.next_inbox_messages` updated to always try to load 1 `ingest-document` related message (`StartIngest` or `CheckFinishIngestDocument`) as a first priority message.
   - This ensures ingest is getting started faster and states are updated frequently (and not a couple of times at the end of ingest)

## Testing notes

Make sure ingest works, same as before (will probably finish a faster, but might only be noticeable with larger ingests)

### CODE REVIEW

File: services/media/service/src/ingest/handlers/check-finish-ingest-document-handler.ts [hunk 1]
```diff
@@ -43,10 +43,30 @@ export class CheckFinishIngestDocumentHandler extends MediaGuardedTransactionalI
     }: TypedTransactionalMessage<CheckFinishIngestDocumentCommand>,
     ownerClient: ClientBase,
   ): Promise<void> {
+    const docId = param(ingest_document_id);
+    await sql`WITH updated AS (
```
Comment (AxTrusov) [inline services/media/service/src/ingest/handlers/check-finish-ingest-document-handler.ts:47]: Checked this query on a ~4500 ingest-items document. It finishes in ~80-90ms. So, I think peformance-wise it should be acceptable.

File: services/media/service/migrations/committed/000026-next-inbox-message-ingest-priority-added.sql [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        NULL;
+    WHEN serialization_failure THEN
+        NULL;
+    WHEN OTHERS THEN
+        RAISE;
+  END;
+
+  -- get (only) the oldest message of every segment but only return it if it is not locked
+  FOR loop_row IN
+    SELECT * FROM app_hidden.inbox m WHERE m.id in (SELECT DISTINCT ON (segment) id
```
Comment (Zehelein) [inline services/media/service/migrations/committed/000026-next-inbox-message-ingest-priority-added.sql]: Can you try combining the query in line 25 with this one e.g. via union? Something like this (not tested):

```
(SELECT * FROM app_hidden.inbox m WHERE m.id in (SELECT DISTINCT ON (segment) id
  FROM app_hidden.inbox
  WHERE processed_at IS NULL AND abandoned_at IS NULL AND aggregate_type = 'ingest-document'
  ORDER BY segment, created_at) order by created_at limit 1)
  
UNION

(SELECT * FROM app_hidden.inbox m WHERE m.id in (SELECT DISTINCT ON (segment) id
  FROM app_hidden.inbox
  WHERE processed_at IS NULL AND abandoned_at IS NULL
  ORDER BY segment, created_at) order by created_at)
```

File: services/media/service/src/ingest/handlers/check-finish-ingest-document-handler.ts (hunk 1 above)
Comment (Zehelein) [reply services/media/service/src/ingest/handlers/check-finish-ingest-document-handler.ts:47]: Can you check that there is a combined index on document_id plus status on the ingest item?

File: services/media/service/src/ingest/handlers/image-succeeded-handler.ts [hunk 3]
```diff
@@ -106,19 +86,16 @@ export abstract class ImageSucceededHandler<
...
+    await update(
+      'ingest_item_steps',
       {
-        ingest_item_step_id: messageContext.ingestItemStepId,
-        ingest_item_id: messageContext.ingestItemId,
-        error_message: getIngestErrorMessage(
+        status: 'ERROR',
+        response_message: getIngestErrorMessage(
           error,
           'An unexpected error occurred while trying to update image relations.',
```
Comment (Zehelein) [inline services/media/service/src/ingest/handlers/image-succeeded-handler.ts:95]: side note: could we maybe update this to something like 

* The image [was correctly imported | already existed] but there was an error adding that image to the entity.

File: services/media/service/src/ingest/handlers/video-succeeded-handler.db.spec.ts [hunk 4]
```diff
@@ -95,7 +84,6 @@ describe('VideoSucceededHandler', () => {
 
   afterEach(async () => {
     await ctx.truncate('ingest_documents');
-    payloads = [];
   });
 
   afterAll(async () => {
```
Comment (Zehelein) [inline services/media/service/src/ingest/handlers/video-succeeded-handler.db.spec.ts:89]: onMessage and onMessageFailure descriptions could be renamed to handleMessage and handleErrorMessage.

File: services/media/service/src/ingest/handlers/image-succeeded-handler.ts (hunk 3 above)
Comment (AxTrusov) [reply services/media/service/src/ingest/handlers/image-succeeded-handler.ts:95]: Will update in context of other upcoming error-handling-related tasks.

File: services/media/service/migrations/committed/000026-next-inbox-message-ingest-priority-added.sql (hunk 2 above)
Comment (AxTrusov) [reply services/media/service/migrations/committed/000026-next-inbox-message-ingest-priority-added.sql]: As discussed, no need to do this. Instead, appropriate messages now have a priority segment assigned to them.

File: services/media/service/src/ingest/handlers/check-finish-ingest-document-handler.ts (hunk 1 above)
Comment (AxTrusov) [reply services/media/service/src/ingest/handlers/check-finish-ingest-document-handler.ts:47]: There was non, only individual indexes. Added.

**Review by Zehelein — APPROVED** [review]

**Pull request merged**